### DISCUSSION TITLE
using Argocd or flux v2 for deploying services

### DISCUSSION THREAD

**Discussion by yasinlachiny** [particuleio/terraform-kubernetes-addons#1456] [General]
Hope you are doing fine.
I'm wondering what is the benefit of deploying services via terragrunt instead of Argocd or flux. We can deploy Flux v2 via this repo and deploy other services via Flux.  Which approach is better? Could you please share your opinion about this idea?

**Comment by ArchiFleKs** [comment]
> Hope you are doing fine. I'm wondering what is the benefit of deploying services via terragrunt instead of Argocd or flux. We can deploy Flux v2 via this repo and deploy other services via Flux. Which approach is better? Could you please share your opinion about this idea?

I'm also a big fan of using Flux to deploy resources. And I'm using it for all the services I'm deploying outside of this repo. The way I see it, this repo handle "complex" addons that needs specific cloud provider permission (like IAM resources on AWS) that you would need to create via TF and then copy paste into your Argo/Flux manifest.

The other way to do it would be to commit file on your Gitops repository directly from Terraform for example.

**Comment by yasinlachiny** [comment]
@ArchiFleKs  Thanks for your response.

I completely agree. 

I also have another problem. There is no guarantee that our master branch and infrastructures are synced. For example, someone can edit something manually. What is the best approach to eliminate drift when we are using this repo?

One simple solution could be to run terragrunt/terraform plan periodically and check drift. Is there any better solution for this kind of issue?

**Reply by ArchiFleKs** [reply]
I personally run terraform in CI system such as Github Action / Gitlab or Atlantis just to name a few.

**Reply by yasinlachiny** [reply]
We are using atlantis too. But atlantis can just deploy our code. it couldn't check drift periodically unless we run it as a cronjob.

**Reply by cloudcompute** [reply]
Hi @ArchiFleKs and @yasinlachiny 

I want to use ArgoCD to deply these addons. What do I need to do replace Flux with ArgoCD.. does there exist any argocd.tf which I can use for reference?

Thank you
